extension/js/common/core/mime.ts

```diff
@@ -403,4 +409,36 @@ export class Mime {
     return node;
   }
 
+  private static extractInlineImagesToAttachments = (html: string, rootNode: any) => {
+    const imgAttNodes: any[] = [];
+    DOMPurify.addHook('afterSanitizeAttributes', (node) => {
+      if (!node) {
+        return;
+      }
+      if ('src' in node) {
+        const img: Element = node;
+        const src = img.getAttribute('src') as string;
+        const { mimeType, data } = Mime.parseInlineImageSrc(src);
+        const imgAttachment = new Attachment({ name: img.getAttribute('name') || '', type: mimeType, data: Buf.fromBase64Str(data), inline: true });
+        const imgAttNode = Mime.createAttNode(imgAttachment);
+        rootNode.appendChild(imgAttNode); // tslint:disable-line:no-unsafe-any
+        const imgAttachmentId = imgAttNode.getHeader('X-Attachment-Id'); // tslint:disable-line:no-unsafe-any
+        img.setAttribute('src', `cid:${imgAttachmentId}`);
+        imgAttNodes.push(imgAttNode);
+      }
+    });
+    const htmlWithInlineImages = DOMPurify.sanitize(html);
+    DOMPurify.removeAllHooks();
+    return htmlWithInlineImages;
+  }
+
+  private static parseInlineImageSrc = (src: string) => {
+    let mimeType;
+    let data = '';
+    const matches = src.match(/data:(image\/\w+);base64,(.*)/);
```

Comment:
This may be a very expensive regex if the image has a few megabytes.

To speed it up, I'd first add `/^` at the beginning, so it doesn't search the whole thing. I'd remove `(.*)` from the end, again so it doesn't need to go that far. If these two are found, then you can extract the data with `.split(',').pop()`.. but parsing is also a source of errors, I think @seisvelas should have a look. Eg if  we do it as I proposed, what if someone produces  `data:image/png;base64,realbase64,fakebase64` causing anomalies.. so you'd have to also check, if you follow this recommendation, how many parts does it have after splitting by `.`

Reply:
Removed the expensive regex and also verified that the format is correct, e.g. there are no unexpected parts.